### ISSUE TITLE
False positive for [CONTENT006] “Broken internal link”

### ISSUE THREAD

**Issue by jan-ferdinand** [joshrotenberg/mdbook-lint#323]
Consider the following section in an mdbook:

```md
## `a_title`

[somewhere](#a_title)
```

Running mdbook-lint incorrectly triggers [CONTENT006]. Perhaps the slugification is incorrect?

On a related note, [CONTENT006] and upward haven’t found their way into the [lint reference](https://github.com/joshrotenberg/mdbook-lint/blob/main/crates/mdbook-lint-cli/example-mdbook-lint.toml) yet.
